kernel/log.h

```diff
@@ -69,6 +70,41 @@ YOSYS_NAMESPACE_BEGIN
 #define S__LINE__sub1(x) S__LINE__sub2(x)
 #define S__LINE__ S__LINE__sub1(__LINE__)
 
+// YS_DEBUGTRAP is a macro that is functionally equivalent to a breakpoint
+// if the platform provides such functionality, and does nothing otherwise.
+// If no debugger is attached, it starts a just-in-time debugger if available,
+// and crashes the process otherwise.
+#if defined(_WIN32)
+# define YS_DEBUGTRAP __debugbreak()
```

Comment:
Is this intrinsic guaranteed to exist for all compilers when `_WIN32` is defined?  Am I wrong, or does this need to be guarded by some checks for its existence like below?

Reply:
> I just wonder about the case of compiling for Windows with a non-MSVC compiler

The [MSDN docs](https://docs.microsoft.com/en-us/cpp/intrinsics/compiler-intrinsics?view=vs-2019) say:

> The use of intrinsics affects the portability of code, because intrinsics that are available in Visual C++ might not be available if the code is compiled with other compilers

In practice, MinGW does support this intrinsic, and Yosys currently doesn't support any other Win32 compilers. Since there is no way to check for an intrinsic other than by adding a guard checking for specific compiler vendors, I would say the guard is redundant and it should be added when (or if) Yosys is ported to another Win32 toolchain, which is in all likelihood will be never.

I would also be fine with using `DebugBreak()` instead.

> Also it looks like you need to `#include <intrin.h>`, too

I missed that, good catch.

Reply:
Aha, I see.  OK, works for me.